platform/iphone/export/export_plugin.cpp

```diff
@@ -728,10 +728,10 @@ Error EditorExportPlatformIOS::_export_loading_screen_images(const Ref<EditorExp
 
 Error EditorExportPlatformIOS::_walk_dir_recursive(DirAccess *p_da, FileHandler p_handler, void *p_userdata) {
 	Vector<String> dirs;
-	String path;
 	String current_dir = p_da->get_current_dir();
 	p_da->list_dir_begin();
-	while ((path = p_da->get_next()).length() != 0) {
+	String path = p_da->get_next();
+	while (path.length() != 0) {
```

Comment:
This isn't necessary strictly speaking as I ended up not changing `String::operator=()`, but I would argue that the change makes the code more readable, even if it does have the drawback that `path = p_da->get_next();` needs to be called manually before each `continue;`.

Reply:
Actually I was able to apply the `operator=()` changes for `String` too, so this is now necessary.